internal/sysfs/poll.go

```diff
@@ -0,0 +1,35 @@
+package sysfs
+
+import (
+	"time"
+
+	"github.com/tetratelabs/wazero/experimental/sys"
+)
+
+// Poll exposes the poll(2) syscall.
+//
+// # Notes on Parameters
+//
+//	We currently expose a pointer to a time.Duration instead of a plain integer.
+//	A nil pointer means "wait forever" whereas poll(2) uses a negative value instead.
+//	This may change in the future.
+//
+// # Notes on the Syscall
```

Comment:
I would use the same doc style as others, but add a special heading
```suggestion
// # Impact of blocking
```